### ISSUE TITLE
ReFine training

### ISSUE THREAD

**Issue by Cupcee** [Wuyxin/ReFine#6]
Hello, thanks for making the explainer available! I was trying to do the training step, and get print like so (running `python refine_train.py --dataset ba3 --hid 50 --epoch 25 --ratio 0.4 --lr 1e-4`):

```
2021-12-22 11:22:08,770 - refine_train.py[line:95] - INFO: number of graphs(train): 2185
2021-12-22 11:22:08,775 - refine_train.py[line:95] - INFO: number of graphs(val):  398
2021-12-22 11:22:08,777 - refine_train.py[line:95] - INFO: number of graphs(test):  397
2021-12-22 11:23:00,702 - refine_train.py[line:158] - INFO: Epoch: 1, LR: 0.00010, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:23:00,709 - refine_train.py[line:163] - INFO: ACC:[[0.31 0.31 0.31 0.32 0.35 0.37 0.52 0.77 0.96 1.  ]] ACC-AUC: 0.521 Mean P: nan
2021-12-22 11:23:49,479 - refine_train.py[line:158] - INFO: Epoch: 2, LR: 0.00010, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:24:37,940 - refine_train.py[line:158] - INFO: Epoch: 3, LR: 0.00010, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:24:37,943 - refine_train.py[line:163] - INFO: ACC:[[0.31 0.31 0.31 0.31 0.34 0.37 0.56 0.76 0.97 1.  ]] ACC-AUC: 0.523 Mean P: nan
2021-12-22 11:25:26,978 - refine_train.py[line:158] - INFO: Epoch: 4, LR: 0.00002, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:26:15,519 - refine_train.py[line:158] - INFO: Epoch: 5, LR: 0.00002, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:26:15,522 - refine_train.py[line:163] - INFO: ACC:[[0.31 0.31 0.31 0.31 0.36 0.37 0.52 0.73 0.97 1.  ]] ACC-AUC: 0.519 Mean P: nan
2021-12-22 11:27:03,904 - refine_train.py[line:158] - INFO: Epoch: 6, LR: 0.00002, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:27:52,747 - refine_train.py[line:158] - INFO: Epoch: 7, LR: 0.00002, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:27:52,750 - refine_train.py[line:163] - INFO: ACC:[[0.31 0.31 0.31 0.31 0.39 0.38 0.52 0.77 0.97 0.99]] ACC-AUC: 0.526 Mean P: nan
2021-12-22 11:28:41,475 - refine_train.py[line:158] - INFO: Epoch: 8, LR: 0.00001, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:29:29,920 - refine_train.py[line:158] - INFO: Epoch: 9, LR: 0.00001, Ratio: 0.40, Train Loss: nan, Val Loss: nan
2021-12-22 11:29:29,923 - refine_train.py[line:163] - INFO: ACC:[[0.31 0.31 0.31 0.31 0.35 0.37 0.51 0.74 0.97 1.  ]] ACC-AUC: 0.517 Mean P: nan
```

Is it normal that the train loss and validation loss are `nan`? Also the ACC-AUC does not seem to meaningfully improve, so I'm wondering where is the issue...

**Comment by Wuyxin** [comment]
Train loss and validation loss are not supposed to be nan, and thus it is reasonable if ACC-AUC is not improved. 

This is something that I never encounter.... So I guess you need to debug to see which step the training went wrong. For example, some embeddings appear to be nan

**Comment by Cupcee** [comment]
Yeah, I thought so. Issue might be with my setup.

**Comment by Wuyxin** [comment]
I think you can start by checking which term of the loss is `nan`, as in
```
loss =  fid_loss + self.gamma * cts_loss
```
Feel free to update your problem. I am curious about how it happened.

**Comment by ZhaoningYu1996** [comment]
Have you solved this problem?
I am also getting similar ACC-AUC results on the mutag dataset.


> Yeah, I thought so. Issue might be with my setup.

**Comment by Cupcee** [comment]
I was not able to solve this yet. Looking at it again now. Did you figure out what was wrong?

**Comment by Cupcee** [comment]
Alright well the issue clearly has something to do with running the code on Google Colab. I tried running it locally on CPU now and the losses look normal.

**Comment by Cupcee** [comment]
Results from running `python ./refine_train.py --dataset ba3 --hid 50 --epoch 25 --ratio 0.4 --lr 1e-4` locally on CPU:
```
2022-01-11 14:06:33,667 - ./refine_train.py[line:160] - INFO: ACC:[[0.31 0.31 0.32 0.33 0.34 0.43 0.58 0.81 0.98 0.99]] ACC-AUC: 
0.539 Mean P: 0.521
2022-01-11 14:07:35,165 - ./refine_train.py[line:158] - INFO: Epoch: 20, LR: 0.00010, Ratio: 0.40, Train Loss: -11.416, Val Loss:
 -0.676
2022-01-11 14:08:36,719 - ./refine_train.py[line:158] - INFO: Epoch: 21, LR: 0.00010, Ratio: 0.40, Train Loss: -12.023, Val Loss:
 -0.646
2022-01-11 14:08:36,721 - ./refine_train.py[line:160] - INFO: ACC:[[0.31 0.31 0.31 0.33 0.35 0.4  0.57 0.8  0.97 0.99]] ACC-AUC: 
0.536 Mean P: 0.525
2022-01-11 14:09:37,821 - ./refine_train.py[line:158] - INFO: Epoch: 22, LR: 0.00010, Ratio: 0.40, Train Loss: -12.633, Val Loss:
 -0.647
2022-01-11 14:10:38,515 - ./refine_train.py[line:158] - INFO: Epoch: 23, LR: 0.00010, Ratio: 0.40, Train Loss: -13.245, Val Loss:
 -0.652
2022-01-11 14:10:38,522 - ./refine_train.py[line:160] - INFO: ACC:[[0.31 0.31 0.31 0.32 0.33 0.42 0.55 0.81 0.97 0.99]] ACC-AUC: 
0.532 Mean P: 0.528
2022-01-11 14:11:41,010 - ./refine_train.py[line:158] - INFO: Epoch: 24, LR: 0.00002, Ratio: 0.40, Train Loss: -13.857, Val Loss:
 -0.675
2022-01-11 14:12:41,682 - ./refine_train.py[line:158] - INFO: Epoch: 25, LR: 0.00002, Ratio: 0.40, Train Loss: -14.464, Val Loss: -0.657
2022-01-11 14:12:41,683 - ./refine_train.py[line:160] - INFO: ACC:[[0.31 0.31 0.32 0.33 0.34 0.42 0.58 0.81 0.98 1.  ]] ACC-AUC: 0.540 Mean P: 0.530

```

ACC-AUC still seems wrong though.

**Comment by Cupcee** [comment]
I was able to get a suitable ACC-AUC 0.624 (and other baselines) by training the indices with a GPU instead of a CPU. I was not aware mapping the GNN from GPU to CPU could make this big of a difference

EDIT: Actually the main cause for this may have been [this](https://github.com/Wuyxin/ReFine/issues/8)

**Comment by smiles724** [comment]
> I was able to get a suitable ACC-AUC 0.624 (and other baselines) by training the indices with a GPU instead of a CPU. I was not aware mapping the GNN from GPU to CPU could make this big of a difference
> 
> EDIT: Actually the main cause for this may have been [this](https://github.com/Wuyxin/ReFine/issues/8)

I also ran it on GPU but was unable to get a decent ACC-AUC of more than 0.6. Can you give me some suggestions on how to fix it?
